unit-tests/sensor/frames/test-filter.cpp

```diff
@@ -0,0 +1,116 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+// We have our own main
+#define CATCH_CONFIG_MAIN
+#define CATCH_CONFIG_RUNNER
+
+#include "../../catch.h"
+
+#include <easylogging++.h>
+#ifdef BUILD_SHARED_LIBS
+// With static linkage, ELPP is initialized by librealsense, so doing it here will
+// create errors. When we're using the shared .so/.dll, the two are separate and we have
+// to initialize ours if we want to use the APIs!
+INITIALIZE_EASYLOGGINGPP
+#endif
+
+#include <librealsense2/rs.hpp>  // Include RealSense Cross Platform API
+#include <chrono>
+#include <atomic>
+
+using namespace std::chrono;
+
+
+// Test description:
+// This test verify that if the user ask for depth stream only, the incoming frames will be depth frames only.
+// L515 device has a frame filter that filters IR frames when depth only is requested
+// Note: * L515 specific test
+//       * Test will pass if no L515 device is connected
+//       * Test will be performed on the first L515 device detected.
+TEST_CASE( "Sensor", "[frames-filter]" )
+{
+    try
+    {
+        static const int FRAMES_TO_CHECK = 100;
+        static const int TEST_TIMEOUT_MS = 10000;
```

Comment:
Why not using chrono?

Reply:
I switched to use chrono for now.
I will create another PR that add a wrapper class to chrono called timeout class and use it in the new PR.